docs: Update modtap docs and README.md (#335)
zmkfirmware/zmk: docs/docs/behavior/mod-tap.md

## Summary

> The Mod-Tap sends a different keypress, depending on if it's held or tapped. If you hold the key for longer than 200ms, the first keycode ("mod") is sent. When you tap the key shortly, the second keycode ("tap") is sent.

Reviewer: Suggestion to improve readability ...
```suggestion
The Mod-Tap behavior sends a different keypress, depending on whether it's held or tapped.
- If you hold the key for longer than 200ms, the first keycode ("mod") is sent.
- If you tap the key (release before 200ms), the second keycode ("tap") is sent.
```
